src/Connector.SqlServer/Connector/SqlServerConnector.cs

```diff
@@ -272,31 +300,30 @@ async Task emptyTable(string name, string context)
                 }
                 catch (Exception e)
                 {
-                    var message = $"Could not empty Container {name}";
+                    var message = $"Could not empty Table {tableName}";
                     _logger.LogError(e, message);
                     throw new CreateContainerException(message);
                 }
             }
-
-            var container = new Container(id, StreamMode);
-            var tasks = new List<Task> { emptyTable(container.PrimaryTable, "Data") };
-            foreach (var table in container.Tables)
-                if (await CheckTableExists(executionContext, providerDefinitionId, table.Value.Name))
-                    tasks.Add(emptyTable(table.Value.Name, table.Key));
-
-            await Task.WhenAll(tasks);
         }
 
         public override async Task ArchiveContainer(ExecutionContext executionContext, Guid providerDefinitionId,
             string id)
         {
             var config = await base.GetAuthenticationDetails(executionContext, providerDefinitionId);
+            var schema = config.GetSchema();
+            var container = new Container(id, StreamMode);
+            var tasks = new List<Task> { archiveTable(container.PrimaryTable, "Data") };
+            foreach (var table in container.Tables)
+                if (await CheckTableExists(executionContext, providerDefinitionId, table.Value.Name.GetValue()))
+                    tasks.Add(archiveTable(table.Value.Name, table.Key));
+
+            await Task.WhenAll(tasks);
 
-            async Task archiveTable(string name, string context)
+            async Task archiveTable(SanitizedSqlName name, string context)
```

Comment:
Agree, same as above
"agree about separate internal methods or even just a straightforward method body instead of local functions. Unfortunately haven't had time to properly refactor this.
At least not in scope of this task."